snapcraft.yaml

```
@@ -915,13 +931,22 @@ parts:
       - lib/*/libnuma*so*
       - lib/*/libpixman*so*
       - lib/*/libusbredir*so*
+      - lib/*/libfdt*.so*
       - share/qemu/keymaps*
-      - share/qemu/efi-virtio.rom*
-      - share/qemu/kvmvapic.bin*
-      - share/qemu/s390-*.img*
-      - share/qemu/slof.bin*
-      - share/qemu/vgabios-*.bin*
       - share/qemu/bios-256k.bin*
+      - share/qemu/efi-virtio*.rom
```

There is only one `efi-virtio` ROM file so I'd use:
```suggestion
      - share/qemu/efi-virtio.rom
```

---

this is to make them option on the non-qemu archs.

---

I think i could further improve the build to not install these packages either on those archs.

---

There's an annoying catch with snapcraft and using selectors (like `on selector1[,selector2]`). The selectors are `AND` instead of `OR`'ed meaning that if you want to make something only for a few arches, you need to duplicate the list like this:

```
build-packages:
  - common-pkg1
  - ...
  - on amd64:
      - virt-enabled-arch-pkg1
      - virt-enabled-arch-pkg2
      - ...
  - on arm64:
      - virt-enabled-arch-pkg1
      - virt-enabled-arch-pkg2
      - ...
  - on s390x:
      - virt-enabled-arch-pkg1
      - virt-enabled-arch-pkg2
      - ...
```

This is explained in https://snapcraft.io/docs/snapcraft-advanced-grammar. That's also the reason we had the `edk2`  part using this:

```
     build-packages:
       - g++
       - on amd64:
         - acpica-tools
         - uuid-dev
       - on arm64:
         - acpica-tools
         - uuid-dev
```
Instead of this:
```
     build-packages:
       - g++
       - on amd64,arm64:
         - acpica-tools
         - uuid-dev
```

So in the end, selectively installing packages for build/stage isn't really convenient :/

---

yeah and it didnt work with the prime section either when i tried it.

---

ill go with the optional indicator instead for prime files